model/post.go

```diff
@@ -292,6 +293,22 @@ func PostPatchFromJson(data io.Reader) *PostPatch {
 	return &post
 }
 
+var channelMentionRegexp = regexp.MustCompile(`(^|\s)(~([a-zA-Z0-9.\-_]*))`)
```

Comment:
The first group can actually be replaced with `\B` since we have partial support for channel linking after punctuation like `~channel-a/~channel-b`. We at least support it on mobile, and it'll be coming to desktop eventually.

Reply:
Actually, we can remove the extra groups to maybe improve performance as well by making this `\B~([a-zA-Z0-9.\-_]*)` or even remove all the groups if we use ``\B~[a-zA-Z0-9.\-_]*` and use a substring to trim off the tilde

Reply:
Updated. With the string `strings.Repeat("this is a ~test message. ", 20)`, before: 46777 ns/op and after: 23521 ns/op.

Reply:
Awesome. Nice to know it actually sped things up